substrate/frame/delegated-staking/src/types.rs

```diff
@@ -131,19 +131,13 @@ impl<T: Config> AgentLedger<T> {
 	///
 	/// Increments provider count if this is a new agent.
 	pub(crate) fn update(self, key: &T::AccountId) {
-		if !<Agents<T>>::contains_key(key) {
-			// This is a new agent. Provide for this account.
-			frame_system::Pallet::<T>::inc_providers(key);
```

Comment:
Agents don't need provider anymore as staking account does not have consumers.
TODO: add migration for agents to dec provider.